osf/management/commands/handle_duplicate_files.py

```diff
@@ -0,0 +1,262 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from osf.models import AbstractNode, Guid
+from addons.osfstorage.models import BaseFileNode
+
+logger = logging.getLogger(__name__)
+
+FOLDER = 'osf.osfstoragefolder'
+
+
+# For a specific file type, returns items that have duplicate
+# target_object_ids, names, parent_id, type, and _path
+# Basically, any items in same location with the same name.
+FETCH_DUPLICATES_BY_FILETYPE = """
+    SELECT *
+    FROM (
+      SELECT
+        target_object_id,
+        name,
+        parent_id,
+        type,
+        _path,
+        count(*) AS ct
+      FROM osf_basefilenode
+      WHERE type = %s
+      AND name != ''
+      GROUP BY (target_object_id, name, parent_id, type, _path)
+    ) as foo
+    WHERE ct > 1;
+    """
+
+# googledrive and trashedfile/folder/folders excluded.
+valid_file_types = [
+    'osf.onedrivefile',
+    'osf.gitlabfile',
+    'osf.dropboxfile',
+    'osf.githubfile',
+    'osf.s3file',
+    'osf.boxfile',
+    'osf.figsharefile',
+    'osf.osfstoragefile',
+    'osf.bitbucketfile',
+    'osf.owncloudfile',
+    FOLDER
+]
+
+def compare_location_and_region(first_file, second_file):
+    """
+    Compares versions for osfstoragefiles -
+    Compares location['bucket'] and location['object'].
+    Also compares version region_ids.
+    """
+    first_loc = first_file['versions__location']
+    second_loc = second_file['versions__location']
+
+    if first_loc and second_loc:
+        return first_loc['bucket'] == second_loc['bucket'] and \
+            first_loc['object'] == second_loc['object'] and \
+            first_file['versions__region_id'] == first_file['versions__region_id']
+    return False
+
+def build_filename_set(files):
+    """
+    Convenience method for building a set of filenames
+    """
+    return set(files.values_list('name', flat=True).order_by('name'))
+
+def compare_basefilenodes(resource_one, resource_two):
+    """
+    Recursively compares the contents of two folders - assumes initial resource_one
+    and resource_two have the same name
+    """
+    match = True
+    item_count = 0
+    files_list_one = BaseFileNode.objects.filter(parent_id=resource_one['id'])
+    files_list_two = BaseFileNode.objects.filter(parent_id=resource_two['id'])
+    if len(files_list_one) != len(files_list_two) or \
+            build_filename_set(files_list_one) != build_filename_set(files_list_two):
+        match = False
+
+    files_list_one = return_basefilenode_values(files_list_one)
+    while match and (item_count < len(files_list_one)):
+        resource_one = files_list_one[item_count]
+        resource_two = return_basefilenode_values(files_list_two.filter(name=resource_one['name'])).first()
+        match = resources_match(resource_one, resource_two) if resource_two else False
+        item_count += 1
+    return match
```

Comment:
3/12 osfstorage folders are not literal duplicates, and the user needs to be contacted. 9/12 are very large `Archive of OSF Storage` folders - probably they're large size is what contributed to something going wrong in the archiving process.